scripts/build_flang.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+import argparse
+import multiprocessing
+import os
+import platform
+import shutil
+import subprocess
+import sys
+from pathlib import Path, PurePosixPath
+
+import settings
+
+
+def default_toolchain():
+    system = platform.uname()[0]
+    machine = platform.uname()[4]
+    toolchain = os.path.join(settings.THIS_DIR,
+                             'cmake',
+                             f'toolchain_{system.lower()}_{machine.lower()}.cmake')
+    return toolchain if os.path.isfile(toolchain) else None
+
+
+def get_arguments():
+    parser = argparse.ArgumentParser(
+        description='Help configure and build Flang')
+
+    buildopt = parser.add_argument_group('general build options')
+    buildopt.add_argument('--builddir', metavar='DIR', default='build',
+                          help='specify build directory (default: "projects_root_dir"/%(default)s) \
+                          Note: The name of the build directory will be the same for flang and libpgmath')
+    buildopt.add_argument('--clean', action='store_true', default=False,
+                          help='clean build')
+    buildopt.add_argument('--build-type', metavar='TYPE', default='Release',
```

Comment:
I think the options should support a single-letter short form, similar to the ones in `build-flang.sh`. `--build-type`, `--install-prefix`, `--target`, and `--use-ccache` are especially important since a user will probably be typing them quite frequently.

Reply:
The options implemented in this script should be synchronized with the options in flang-compiler/classic-flang-llvm-project#134.

Reply:
> I think the options should support a single-letter short form, similar to the ones in `build-flang.sh`. `--build-type`, `--install-prefix`, `--target`, and `--use-ccache` are especially important since a user will probably be typing them quite frequently.

@bryanpkc Thanks for the review. That's good point.  I'm adding a new singel-letter form to the options.

Reply:
@bryanpkc  I've updated the build script.

- Added single-letter support for --builddir (-b), --debug (-d), --target (-t), --use-ccache (-c), -- install-prefix (-p)
- Removed superfluous  alias options such as build_type, llvm-root
- Refactor the path handling: os.path is replaced with pathlib, now the path handling is consistent.

Reply:
I suggest you revert the `--debug | -d` change, and keep the `--build-type | -b` option, so that a user can specify a `RelWithDebInfo` build or even `MinSizeRel` build. You can use the `--build-dir | -d` option to specify the location of the build. I'd also suggest adding a `--verbose | -v` option so that `ninja`, `make`, or `nmake` can be invoked with an appropriate option to print verbose messages.